packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3169,6 +3169,23 @@ export class ContainerRuntime
 		return { latestSnapshotRefSeq, latestSnapshotVersionId: versionId };
 	}
 
+	private async failOrFetchLatestSnapshotFromStorage(
```

Comment:
A new function should not be needed. You can simply move the check and failure into `fetchSnapshotFromStorage` (not `fetchLatestSnapshotFromStorage`) function.